src/rules/lowercase-name.ts

```diff
@@ -96,13 +103,18 @@ export default createRule({
             items: { enum: ['describe', 'test', 'it'] },
             additionalItems: false,
           },
+          allow: {
+            type: 'array',
+            items: { type: 'string' },
+            additionalItems: false,
+          },
         },
         additionalProperties: false,
       },
     ],
   } as const,
-  defaultOptions: [{ ignore: [] } as { ignore: readonly JestFunctionName[] }],
-  create(context, [{ ignore }]) {
+  defaultOptions: [{ ignore: [], allow: [] }],
+  create(context, [{ ignore = [], allow = [] }]) {
```

Comment:
```suggestion
  create(context, [{ ignore, allow }]) {
```

`defaultOptions` should do it for you

Reply:
Agreed, but TypeScript doesn't know that :(